tests/50federation/34room-backfill.pl

```diff
@@ -0,0 +1,168 @@
+use Future::Utils qw( repeat );
+
+test "Outbound federation can backfill events",
+   requires => [ local_user_fixture(), $main::INBOUND_SERVER, federation_user_id_fixture() ],
+
+   do => sub {
+      my ( $user, $inbound_server, $creator_id ) = @_;
+
+      my $local_server_name = $inbound_server->server_name;
+      my $datastore         = $inbound_server->datastore;
+
+      my $room_alias = "#50fed-31backfill:$local_server_name";
+
+      my $room = $datastore->create_room(
+         creator => $creator_id,
+         alias   => $room_alias,
+      );
+
+      # Create some past messages to backfill from
+      $room->create_event(
+         type => "m.room.message",
+
+         sender  => $creator_id,
+         content => {
+            msgtype => "m.text",
+            body    => "Message $_ here",
+         },
+      ) for 1 .. 10;
+
+      Future->needs_all(
+         $inbound_server->await_backfill( $room->room_id )->then( sub {
+            my ( $req ) = @_;
+
+            my $v     = $req->query_param( 'v' );
```

Comment:
Nope, more please :) Maybe you could give some examples like "these may be the IDs of the events the other server has most recently seen"?